### ISSUE TITLE
replace optparse with argparse for 2.7 and 3.2 builds

### ISSUE THREAD

**Issue by ajf58** [ajf58/tigris-sandbox#2763]
This issue was originally created at: 2011-05-14 12:03:44.
This issue was reported by: `ipatrol`.
ipatrol said at 2011-05-14 12:03:44
>Argparse is more flexible and can be easily configured to use various command 
>conventions, and its type system makes integration with SCons much easier. I am 
>willing to provide a patch myself if no one has any objections to the change.

ipatrol said at 2011-05-14 12:29:10
>Here are some open bugs that either demonstrate obstacles to implementation or 
>themselves would be fixed if this change is implemented.

garyo said at 2011-05-15 05:13:36
>tweak deps

garyo said at 2011-05-15 05:14:11
>grr

garyo said at 2011-05-15 05:14:53
>sorry about that, putting deps back.

garyo said at 2011-05-15 05:21:58
>This is a good idea and a patch would be welcome; however back-compatibility is 
>important for SCons.  SCons needs to be compatible back to 2.3, so either argparse 
>has to be back-ported and included in the SCons.compat package (any idea how easy 
>that would be?) or we'd have to wait to include it until a future release which 
>would only be compatible back to 2.7, which may be quite some time.

ipatrol said at 2011-05-15 08:09:30
>If you look on PyPi, there is an argparse backport, but my original suggestion was to only use 
>argparse, but my original suggestion was to only make the change for 2.7<= builds. Hence I am 
>adding a floor2.5 keyword because the difficulties of application increase exponentially with each 
>version. I myself would consider it prudent to terminate support to 2.3 and 2.4, but if there are 
>problems with that, pypreprocessor can allow variants of code to execute on different builds for 
>different versions. I am already working on the rewrite though, it should be simple.

dirkbaechle said at 2014-04-27 03:52:51
>*** Issue 2946 has been marked as a duplicate of this issue. ***

More information about this issue is at http://docs.python.org/library/argparse.html.

ipatrol said this issue depends on #2064 at 2011-05-14 12:29:10.

ipatrol said this issue depends on #2748 at 2011-05-14 12:29:10.

ipatrol said this issue blocks #2064 at 2011-05-14 12:29:10.

ipatrol said this issue blocks #2748 at 2011-05-14 12:29:10.

ipatrol said this issue  is a duplicate of #2064 at 2011-05-14 12:29:10.

ipatrol said this issue  is a duplicate of #2748 at 2011-05-14 12:29:10.

ipatrol said this issue is duplicated by #2064 at 2011-05-14 12:29:10.

ipatrol said this issue is duplicated by #2748 at 2011-05-14 12:29:10.
